crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1101 @@
+            for system in conditions.borrow_mut().iter_mut() {
+                system.check_change_tick(change_tick);
+            }
+        }
+
+        for conditions in &mut self.executable.set_conditions {
+            for system in conditions.borrow_mut().iter_mut() {
+                system.check_change_tick(change_tick);
+            }
+        }
+    }
+}
+
+/// A directed acylic graph structure.
+#[derive(Default)]
+struct Dag {
+    /// A directed graph.
+    graph: DiGraphMap<NodeId, ()>,
+    /// A cached topological ordering of the graph.
+    topsort: Vec<NodeId>,
+}
+
+impl Dag {
+    fn new() -> Self {
+        Self {
+            graph: DiGraphMap::new(),
+            topsort: Vec::new(),
+        }
+    }
+}
+
+/// A [`SystemSet`] with metadata, stored in a [`ScheduleGraph`].
+struct SystemSetNode {
+    inner: BoxedSystemSet,
+    /// `true` if this system set was modified with `configure_set`
+    configured: bool,
+}
+
+impl SystemSetNode {
+    pub fn new(set: BoxedSystemSet) -> Self {
+        Self {
+            inner: set,
+            configured: false,
+        }
+    }
+
+    pub fn name(&self) -> String {
+        format!("{:?}", &self.inner)
+    }
+
+    pub fn is_system_type(&self) -> bool {
+        self.inner.is_system_type()
+    }
+}
+
+/// Metadata for a [`Schedule`].
+#[derive(Default)]
+struct ScheduleGraph {
+    systems: Vec<Option<BoxedSystem>>,
+    system_conditions: Vec<Option<Vec<BoxedCondition>>>,
+    system_sets: Vec<SystemSetNode>,
+    system_set_conditions: Vec<Option<Vec<BoxedCondition>>>,
+    system_set_ids: HashMap<BoxedSystemSet, NodeId>,
+    uninit: Vec<(NodeId, usize)>,
+    hierarchy: Dag,
+    dependency: Dag,
+    dependency_flattened: Dag,
+    ambiguous_with: UnGraphMap<NodeId, ()>,
+    ambiguous_with_flattened: UnGraphMap<NodeId, ()>,
+    ambiguous_with_all: HashSet<NodeId>,
+    default_set: Option<BoxedSystemSet>,
+    changed: bool,
+    settings: ScheduleBuildSettings,
+}
+
+impl ScheduleGraph {
+    pub fn new() -> Self {
+        Self {
+            systems: Vec::new(),
+            system_conditions: Vec::new(),
+            system_sets: Vec::new(),
+            system_set_conditions: Vec::new(),
+            system_set_ids: HashMap::new(),
+            uninit: Vec::new(),
+            hierarchy: Dag::new(),
+            dependency: Dag::new(),
+            dependency_flattened: Dag::new(),
+            ambiguous_with: UnGraphMap::new(),
+            ambiguous_with_flattened: UnGraphMap::new(),
+            ambiguous_with_all: HashSet::new(),
+            default_set: None,
+            changed: false,
+            settings: default(),
+        }
+    }
+
+    fn set_default_set(&mut self, set: impl SystemSet) {
+        assert!(!set.is_system_type(), "invalid use of system type set");
+        self.default_set = Some(Box::new(set));
+    }
+
+    fn set_build_settings(&mut self, settings: ScheduleBuildSettings) {
```

Comment:
Given that this method isn't at all pub, I don't see what value a setter method is here.